runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/WindmillSink.java

```diff
@@ -75,11 +78,20 @@ private static ByteString encodeMetadata(
       ByteStringOutputStream stream,
       Coder<Collection<? extends BoundedWindow>> windowsCoder,
       Collection<? extends BoundedWindow> windows,
-      PaneInfo paneInfo)
+      PaneInfo paneInfo,
+      BeamFnApi.Elements.ElementMetadata metadata)
       throws IOException {
     try {
-      PaneInfoCoder.INSTANCE.encode(paneInfo, stream);
-      windowsCoder.encode(windows, stream, Coder.Context.OUTER);
+      // element metadata is behind the experiment
+      boolean elementMetadata = WindowedValues.WindowedValueCoder.isMetadataSupported();
+      if (elementMetadata) {
+        PaneInfoCoder.INSTANCE.encode(paneInfo, stream);
```

Comment:
IIRC in the design there would be a bit in the PaneInfo that indicates if there is any metadata in the stream, right? Or I guess that is only for the coder and we always put it in the WindmillSink/WindmillSource exchange?